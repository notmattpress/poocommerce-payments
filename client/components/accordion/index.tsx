/**
 * Internal dependencies
 */
import Accordion from './accordion';
import AccordionBody from './body';
import AccordionRow from './row';

export { Accordion, AccordionBody, AccordionRow };
