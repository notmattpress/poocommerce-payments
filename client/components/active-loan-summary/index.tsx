/**
 * External dependencies
 */
import * as React from 'react';
import {
	Button,
	Card,
	CardBody,
	CardHeader,
	Flex,
	FlexBlock,
	FlexItem,
} from '@wordpress/components';
import { __, sprintf } from '@wordpress/i18n';
import { createInterpolateElement } from '@wordpress/element';

/**
 * Internal dependencies.
 */
import { formatExplicitCurrency } from 'multi-currency/interface/functions';
import Loadable from 'components/loadable';
import { useActiveLoanSummary } from 'wcpay/data';
import { getAdminUrl } from 'wcpay/utils';

import './style.scss';
import { formatDateTimeFromTimestamp } from 'wcpay/utils/date-time';

const Block = ( {
	title,
	children,
}: {
	title: React.ReactNode;
	children: React.ReactNode;
} ): JSX.Element => (
	<FlexBlock className="wcpay-loan-summary-block">
		<div className="wcpay-loan-summary-block__title">{ title }</div>
		<div className="wcpay-loan-summary-block__value">{ children }</div>
	</FlexBlock>
);

const ActiveLoanSummaryLoading = (): JSX.Element => {
	return (
		<Card>
			<CardHeader size="medium" className="wcpay-loan-summary-header">
				<FlexItem>
					{ __( 'Active loan overview', 'poocommerce-payments' ) }
				</FlexItem>
			</CardHeader>
			<CardBody className="wcpay-loan-summary-body">
				<Flex align="normal" className="wcpay-loan-summary-row">
					<Block
						title={ __( 'Total repaid', 'poocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder={ __(
								'Total repaid placeholder',
								'poocommerce-payments'
							) }
						/>
					</Block>
					<Block
						title={ __(
							'Repaid this period',
							'poocommerce-payments'
						) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder={ __(
								'Repaid this period placeholder',
								'poocommerce-payments'
							) }
						/>
					</Block>
				</Flex>
				<Flex
					align="normal"
					className="wcpay-loan-summary-row is-bottom-row"
				>
					<Block
						title={ __( 'Loan disbursed', 'poocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Date disbursed"
						/>
					</Block>
					<Block
						title={ __( 'Loan amount', 'poocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Loan amount"
						/>
					</Block>
					<Block title={ __( 'Fixed fee', 'poocommerce-payments' ) }>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Fixed fee"
						/>
					</Block>
					<Block
						title={ __( 'Withhold rate', 'poocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Rate"
						/>
					</Block>
					<Block
						title={ __( 'First paydown', 'poocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder={ __(
								'First paydown',
								'poocommerce-payments'
							) }
						/>
					</Block>
				</Flex>
			</CardBody>
		</Card>
	);
};

const getActiveLoanId = () => {
	for ( const i in wcpaySettings.accountLoans.loans ) {
		const [ loanId, status ] = wcpaySettings.accountLoans.loans[ i ].split(
			'|'
		);
		if ( 'active' === status ) {
			return loanId;
		}
	}
	return null;
};

const ActiveLoanSummary = (): JSX.Element => {
	const { summary, isLoading } = useActiveLoanSummary();

	if ( isLoading || ! summary ) {
		return <ActiveLoanSummaryLoading />;
	}

	const { details } = summary;

	return (
		<Card>
			<CardHeader size="medium" className="wcpay-loan-summary-header">
				<FlexItem>
					{ __( 'Active loan overview', 'poocommerce-payments' ) }
				</FlexItem>
				<FlexItem>
					{ getActiveLoanId() && (
						<Button
							isLink
							href={ getAdminUrl( {
								page: 'wc-admin',
								path: '/payments/transactions',
								type: 'charge',
								filter: 'advanced',
								loan_id_is: getActiveLoanId(),
							} ) }
						>
							{ __(
								'View transactions',
								'poocommerce-payments'
							) }
						</Button>
					) }
				</FlexItem>
			</CardHeader>
			<CardBody className="wcpay-loan-summary-body">
				<Flex align="normal" className="wcpay-loan-summary-row">
					<Block
						title={ __( 'Total repaid', 'poocommerce-payments' ) }
					>
						{ createInterpolateElement(
							sprintf(
								__(
									'<big>%s</big> of %s',
									'poocommerce-payments'
								),
								formatExplicitCurrency(
									details.paid_amount,
									details.currency
								),
								formatExplicitCurrency(
									details.fee_amount + details.advance_amount,
									details.currency
								)
							),
							{
								big: <span className="is-big" />,
							}
						) }
					</Block>
					<Block
						title={ sprintf(
							__(
								'Repaid this period (until %s)',
								'poocommerce-payments'
							),
							formatDateTimeFromTimestamp(
								details.current_repayment_interval.due_at
							)
						) }
					>
						{ createInterpolateElement(
							sprintf(
								__(
									'<big>%s</big> of %s minimum',
									'poocommerce-payments'
								),
								formatExplicitCurrency(
									details.current_repayment_interval
										.paid_amount,
									details.currency
								),
								formatExplicitCurrency(
									details.current_repayment_interval
										.paid_amount +
										details.current_repayment_interval
											.remaining_amount,
									details.currency
								)
							),
							{
								big: <span className="is-big" />,
							}
						) }
					</Block>
				</Flex>
				<Flex
					align="normal"
					className="wcpay-loan-summary-row is-bottom-row"
				>
					<Block
						title={ __( 'Loan disbursed', 'poocommerce-payments' ) }
					>
						{ formatDateTimeFromTimestamp(
							details.advance_paid_out_at
						) }
					</Block>
					<Block
						title={ __( 'Loan amount', 'poocommerce-payments' ) }
					>
						{ formatExplicitCurrency(
							details.advance_amount,
							details.currency
						) }
					</Block>
					<Block title={ __( 'Fixed fee', 'poocommerce-payments' ) }>
						{ formatExplicitCurrency(
							details.fee_amount,
							details.currency
						) }
					</Block>
					<Block
						title={ __( 'Withhold rate', 'poocommerce-payments' ) }
					>
						{ details.withhold_rate * 100 }%
					</Block>
					<Block
						title={ __( 'First paydown', 'poocommerce-payments' ) }
					>
						{ formatDateTimeFromTimestamp(
							details.repayments_begin_at
						) }
					</Block>
				</Flex>
			</CardBody>
		</Card>
	);
};

export default ActiveLoanSummary;
