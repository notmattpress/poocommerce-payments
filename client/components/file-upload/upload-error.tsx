/** @format **/

/**
 * External dependencies
 */
import React from 'react';

interface UploadErrorProps {
	error?: string;
}

const FileUploadError = ( { error }: UploadErrorProps ): JSX.Element => {
	return <span className="upload-message is-destructive">{ error }</span>;
};

export default FileUploadError;
