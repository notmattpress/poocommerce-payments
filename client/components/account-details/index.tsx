/** @format **/

/**
 * External dependencies
 */
import React from 'react';
import { __ } from '@wordpress/i18n';
import { addQueryArgs } from '@wordpress/url';
import { Card, CardBody, CardHeader, Flex } from '@wordpress/components';

/**
 * Internal dependencies
 */
import './style.scss';
import {
	AccountDetailsType,
	AccountDetailsData,
} from 'wcpay/types/account/account-details';
import { AccountTools } from 'wcpay/components/account-status/account-tools';
import AccountFees from 'wcpay/components/account-status/account-fees';
import Banner from './banner';
import PayoutStatusWrapper from './payout-status-wrapper';
import HeaderTitle from './header-title';

interface AccountDetailsProps {
	accountDetails: AccountDetailsType;
	accountFees?: any[];
	accountLink?: string;
}

const AccountDetailsCard: React.FC< {
	title: React.ReactNode;
	children: React.ReactNode;
} > = ( { title, children } ) => {
	return (
		<Card size="medium">
			<CardHeader className="woopayments-account-details__header">
				{ title }
			</CardHeader>
			<CardBody>{ children }</CardBody>
		</Card>
	);
};

const AccountDetailsError: React.FC = () => {
	const cardTitle = __( 'Account details', 'poocommerce-payments' );
	return (
		<AccountDetailsCard title={ cardTitle }>
			{ __( 'Error loading account details.', 'poocommerce-payments' ) }
		</AccountDetailsCard>
	);
};

const AccountDetailsContent: React.FC< {
	accountDetails: AccountDetailsData;
	accountFees: any[];
	accountLink?: string;
} > = ( { accountDetails, accountFees, accountLink } ) => {
	const processedAccountLink = accountLink
		? addQueryArgs( accountLink, {
				from: 'WCPAY_ACCOUNT_DETAILS',
				source: 'wcpay-account-details',
		  } )
		: null;

	const cardTitle = (
		<>
			<HeaderTitle
				accountStatus={ accountDetails.account_status }
				accountLink={ processedAccountLink }
			/>
			<Banner banner={ accountDetails.banner } />
		</>
	);

	return (
		<AccountDetailsCard title={ cardTitle }>
			<PayoutStatusWrapper
				payoutStatus={ accountDetails.payout_status }
			/>

			<AccountTools />

			{ accountFees && accountFees.length > 0 && (
				<AccountFees accountFees={ accountFees } />
			) }
		</AccountDetailsCard>
	);
};

const AccountDetails: React.FC< AccountDetailsProps > = ( {
	accountDetails,
	accountFees = [],
	accountLink,
} ) => {
	return null === accountDetails ? (
		<AccountDetailsError />
	) : (
		<AccountDetailsContent
			accountDetails={ accountDetails }
			accountFees={ accountFees }
			accountLink={ accountLink }
		/>
	);
};

export default AccountDetails;
