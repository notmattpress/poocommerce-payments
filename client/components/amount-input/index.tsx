/**
 * External dependencies
 */
import React, { useCallback, useEffect } from 'react';
// @ts-expect-error The component exists.
import { TextControlWithAffixes } from '@poocommerce/components';
import './style.scss';

interface AmountInputProps {
	id?: string;
	label?: string;
	prefix?: string;
	value: string;
	placeholder?: string;
	help?: string;
	onChange?: ( value: string ) => void;
}

const AmountInput: React.FC< AmountInputProps > = ( {
	id,
	prefix,
	label,
	value,
	placeholder,
	help,
	onChange = () => null,
} ) => {
	// Only allow digits, a single dot, and more digits (or an empty value).
	const validateInput = useCallback(
		( subject: string ) => /^(\d+\.?\d*)?$/m.test( subject ),
		[]
	);

	const validatedValue = validateInput( value ) ? value : '';

	const [ internalValue, setInternalValue ] = React.useState(
		validatedValue
	);

	useEffect( () => {
		if ( ! validateInput( internalValue ) ) {
			onChange( '' );
		}
	}, [ validateInput, internalValue, onChange ] );

	const handleChange = ( inputValue: string ) => {
		if ( validateInput( inputValue ) ) {
			setInternalValue( inputValue );
			onChange( inputValue );
		}
	};

	return (
		<TextControlWithAffixes
			id={ id }
			className="wcpay-components-amount-input"
			label={ label }
			prefix={ prefix }
			value={ internalValue }
			data-testid="amount-input"
			onChange={ handleChange }
			placeholder={ placeholder }
			help={ help }
		/>
	);
};

export default AmountInput;
