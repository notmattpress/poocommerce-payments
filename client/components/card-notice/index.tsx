/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import './styles.scss';
import { CardFooter } from 'wcpay/components/wp-components-wrapped/components/card-footer';

interface CardNoticeProps {
	children: React.ReactNode;
	actions?: JSX.Element;
}

const CardNotice: React.FC< React.PropsWithChildren< CardNoticeProps > > = ( {
	children,
	actions,
} ) => {
	return (
		<CardFooter className="card-notice">
			<div className="card-notice__section">
				<div className="card-notice__text">{ children }</div>
				<div className="card-notice__button">{ actions }</div>
			</div>
		</CardFooter>
	);
};

export default CardNotice;
