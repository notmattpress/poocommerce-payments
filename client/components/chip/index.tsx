/** @format **/

/**
 * External dependencies
 */
import React from 'react';
import { Icon } from '@wordpress/icons';
import clsx from 'clsx';

/**
 * Internal dependencies
 */
import './style.scss';
import { HoverTooltip } from 'wcpay/components/tooltip';

export type ChipType = 'primary' | 'success' | 'light' | 'warning' | 'alert';

interface Props {
	message: string;
	type?: ChipType;
	className?: string;
	tooltip?: string;
	icon?: React.JSX.Element;
}
const Chip: React.FC< React.PropsWithChildren< Props > > = ( {
	message,
	type = 'primary',
	className,
	tooltip,
	icon,
} ) => {
	const classNames = clsx(
		'chip',
		`chip-${ type }`,
		icon && 'chip-has-icon',
		className
	);

	const content = (
		<>
			{ icon && <Icon icon={ icon } size={ 16 } /> }
			{ message }
		</>
	);

	if ( tooltip ) {
		return (
			<HoverTooltip content={ tooltip }>
				<span className={ classNames }>{ content }</span>
			</HoverTooltip>
		);
	}
	return <span className={ classNames }>{ content }</span>;
};

export default Chip;
