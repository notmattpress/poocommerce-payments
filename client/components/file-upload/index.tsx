/**
 * External dependencies
 */
import React from 'react';
import { __ } from '@wordpress/i18n';
import CheckmarkIcon from 'gridicons/dist/checkmark';
import ImageIcon from 'gridicons/dist/image';
import AddOutlineIcon from 'gridicons/dist/add-outline';
import TrashIcon from 'gridicons/dist/trash';
import { DropZone, FormFileUpload, Button } from '@wordpress/components';

/**
 * Internal dependencies.
 */
import FileUploadError from './upload-error';
import FileUploadPreview from './preview';

interface FileUploadControlProps {
	fieldKey: string;
	fileName: string;
	disabled?: boolean;
	isDone: boolean;
	isLoading: boolean;
	accept: string;
	error?: string;
	onFileChange( key: string, file: File ): Promise< void >;
	onFileRemove( key: string ): void;
	help?: string;
	showPreview?: boolean;
	uploadButtonLabel?: string;
	type?: string;
}

export const FileUploadControl = ( {
	fieldKey,
	fileName,
	disabled,
	isDone,
	isLoading,
	accept,
	error,
	onFileChange,
	onFileRemove,
	showPreview,
	uploadButtonLabel,
	type = 'file',
}: FileUploadControlProps ): JSX.Element => {
	const hasError = ( error && 0 < error.length ) || false;

	const IconType = type === 'image' ? ImageIcon : AddOutlineIcon;
	const Icon = isDone && ! hasError ? CheckmarkIcon : IconType;

	const handleButtonClick = (
		event: React.MouseEvent< HTMLButtonElement >,
		openFileDialog: () => void
	) => {
		// Get file input next to the button element and clear it's value,
		// allowing to select the same file again in case of
		// connection or general error or just need to select it again.
		// This workaround is useful until we update @wordpress/components to a
		// version the supports this: https://github.com/WordPress/gutenberg/issues/39267
		const fileInput:
			| HTMLInputElement
			| null
			| undefined = ( event.target as HTMLButtonElement )
			.closest( '.components-form-file-upload' )
			?.querySelector( 'input[type="file"]' );

		if ( fileInput ) {
			fileInput.value = '';
		}

		openFileDialog();
	};

	return (
		<>
			<DropZone
				onFilesDrop={ ( files: Array< File > ) =>
					onFileChange( fieldKey, files[ 0 ] )
				}
			/>
			<div className="file-upload">
				<FormFileUpload
					accept={ accept }
					onChange={ (
						event: React.ChangeEvent< HTMLInputElement >
					): void => {
						onFileChange(
							fieldKey,
							( event.target.files || new FileList() )[ 0 ]
						);
					} }
					render={ ( { openFileDialog } ) => (
						<Button
							id={ `form-file-upload-${ fieldKey }` }
							className={
								isDone && ! hasError ? 'is-success' : ''
							}
							isSecondary
							isDestructive={ hasError }
							isBusy={ isLoading }
							disabled={ disabled || isLoading }
							icon={ <Icon size={ 18 } /> }
							onClick={ (
								event: React.MouseEvent< HTMLButtonElement >
							) => handleButtonClick( event, openFileDialog ) }
						>
							{ uploadButtonLabel ||
								__( 'Upload file', 'woocommerce-payments' ) }
						</Button>
					) }
				/>

				{ hasError ? (
					<FileUploadError error={ error } />
				) : (
					<FileUploadPreview
						fileName={ fileName }
						showPreview={ showPreview }
					/>
				) }

				{ isDone && ! disabled ? (
					<Button
						className="delete-uploaded-file-button"
						aria-label={ __(
							'Remove file',
							'woocommerce-payments'
						) }
						icon={ <TrashIcon size={ 18 } /> }
						onClick={ () => onFileRemove( fieldKey ) }
					/>
				) : null }
			</div>
		</>
	);
};
