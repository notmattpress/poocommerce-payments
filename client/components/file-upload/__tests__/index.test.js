/** @format */

/**
 * External dependencies
 */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { FileUploadControl } from 'components/file-upload';

describe( 'FileUploadControl', () => {
	const accept = '.pdf, image/png, image/jpeg';
	let baseProps;

	beforeEach( () => {
		baseProps = {
			fieldKey: 'field_key',
			accept,
			isDone: false,
			isLoading: false,
			fileName: '',
			error: '',
			disabled: false,
			onFileChange: jest.fn(),
			onFileRemove: jest.fn(),
		};

		global.wcpaySettings = {
			restUrl: 'http://example.com/wp-json/',
		};
	} );

	test( 'renders default file upload control', () => {
		const { container: control } = render(
			<FileUploadControl { ...baseProps } />
		);
		expect( control ).toMatchSnapshot();
	} );

	test( 'renders loading state', () => {
		const { container: control } = render(
			<FileUploadControl { ...baseProps } isLoading={ true } />
		);
		expect( control ).toMatchSnapshot();
	} );

	test( 'renders upload done state', () => {
		const { container: control } = render(
			<FileUploadControl
				{ ...baseProps }
				isDone={ true }
				fileName="file.pdf"
			/>
		);
		expect( control ).toMatchSnapshot();
	} );

	test( 'renders upload failed state', () => {
		const { container: control } = render(
			<FileUploadControl
				{ ...baseProps }
				error="Error message"
				fileName="file.pdf"
			/>
		);
		expect( control ).toMatchSnapshot();
	} );

	test( 'triggers onFileChange', () => {
		const { container: control } = render(
			<FileUploadControl { ...baseProps } />
		);
		const fakeFile = {};
		const fakeEvent = { target: { files: [ fakeFile ] } };

		// Note: FormFileUpload does not associate file input with label so workaround is required to select it.
		const input = control.querySelector( 'input[type="file"]' );
		if ( input !== null ) {
			fireEvent.change( input, fakeEvent );
		}

		expect( baseProps.onFileChange ).toHaveBeenCalledTimes( 1 );
		expect( baseProps.onFileChange ).toHaveBeenCalledWith(
			'field_key',
			fakeFile
		);
	} );

	test( 'triggers onFileChange two times when selecting the same file again', () => {
		const { container: control } = render(
			<FileUploadControl { ...baseProps } />
		);

		const file = new File( [ 'hello' ], 'hello.png', {
			type: 'image/png',
		} );

		// Note: FormFileUpload does not associate file input with label so workaround is required to select it.
		const input = control.querySelector( 'input[type="file"]' );
		if ( input !== null ) {
			fireEvent.change( input, { target: { files: [ file ] } } );
			fireEvent.change( input, { target: { files: [ file ] } } );
		}

		expect( baseProps.onFileChange ).toHaveBeenNthCalledWith(
			2,
			'field_key',
			file
		);
	} );

	test( 'triggers onFileRemove', () => {
		const { getByRole } = render(
			<FileUploadControl
				{ ...baseProps }
				fileName="file.pdf"
				isDone={ true }
			/>
		);
		fireEvent.click( getByRole( 'button', { name: /remove file/i } ) );
		expect( baseProps.onFileRemove ).toHaveBeenCalledTimes( 1 );
		expect( baseProps.onFileRemove ).toHaveBeenCalledWith( 'field_key' );
	} );

	test( 'renders disabled state', () => {
		const { container: control } = render(
			<FileUploadControl
				{ ...baseProps }
				disabled={ true }
				isDone={ true }
				fileName="file.pdf"
			/>
		);
		expect( control ).toMatchSnapshot();
	} );
} );
