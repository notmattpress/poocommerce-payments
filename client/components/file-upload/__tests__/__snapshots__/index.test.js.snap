// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploadControl renders default file upload control 1`] = `
<div>
  <div
    class="components-drop-zone"
    data-is-drop-zone="true"
  >
    <div
      class="components-drop-zone__content"
    >
      <div
        class="components-drop-zone__content-inner"
      >
        <svg
          aria-hidden="true"
          class="components-drop-zone__content-icon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.5 15v3.5H13V6.7l4.5 4.1 1-1.1-6.2-5.8-5.8 5.8 1 1.1 4-4v11.7h-6V15H4v5h16v-5z"
          />
        </svg>
        <span
          class="components-drop-zone__content-text"
        >
          Drop files to upload
        </span>
      </div>
    </div>
  </div>
  <div
    class="file-upload"
  >
    <div
      class="components-form-file-upload"
    >
      <button
        class="components-button is-secondary has-text has-icon"
        id="form-file-upload-field_key"
        type="button"
      >
        <svg
          class="gridicon gridicons-add-outline"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm5 9h-4V7h-2v4H7v2h4v4h2v-4h4v-2z"
            />
          </g>
        </svg>
        Upload file
      </button>
      <input
        accept=".pdf, image/png, image/jpeg"
        data-testid="form-file-upload-input"
        style="display: none;"
        type="file"
      />
    </div>
  </div>
</div>
`;

exports[`FileUploadControl renders disabled state 1`] = `
<div>
  <div
    class="components-drop-zone"
    data-is-drop-zone="true"
  >
    <div
      class="components-drop-zone__content"
    >
      <div
        class="components-drop-zone__content-inner"
      >
        <svg
          aria-hidden="true"
          class="components-drop-zone__content-icon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.5 15v3.5H13V6.7l4.5 4.1 1-1.1-6.2-5.8-5.8 5.8 1 1.1 4-4v11.7h-6V15H4v5h16v-5z"
          />
        </svg>
        <span
          class="components-drop-zone__content-text"
        >
          Drop files to upload
        </span>
      </div>
    </div>
  </div>
  <div
    class="file-upload"
  >
    <div
      class="components-form-file-upload"
    >
      <button
        class="components-button is-success is-secondary has-text has-icon"
        disabled=""
        id="form-file-upload-field_key"
        type="button"
      >
        <svg
          class="gridicon gridicons-checkmark"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M9 19.414l-6.707-6.707 1.414-1.414L9 16.586 20.293 5.293l1.414 1.414z"
            />
          </g>
        </svg>
        Upload file
      </button>
      <input
        accept=".pdf, image/png, image/jpeg"
        data-testid="form-file-upload-input"
        style="display: none;"
        type="file"
      />
    </div>
    <span
      class="upload-message"
    >
      file.pdf
    </span>
  </div>
</div>
`;

exports[`FileUploadControl renders loading state 1`] = `
<div>
  <div
    class="components-drop-zone"
    data-is-drop-zone="true"
  >
    <div
      class="components-drop-zone__content"
    >
      <div
        class="components-drop-zone__content-inner"
      >
        <svg
          aria-hidden="true"
          class="components-drop-zone__content-icon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.5 15v3.5H13V6.7l4.5 4.1 1-1.1-6.2-5.8-5.8 5.8 1 1.1 4-4v11.7h-6V15H4v5h16v-5z"
          />
        </svg>
        <span
          class="components-drop-zone__content-text"
        >
          Drop files to upload
        </span>
      </div>
    </div>
  </div>
  <div
    class="file-upload"
  >
    <div
      class="components-form-file-upload"
    >
      <button
        class="components-button is-secondary is-busy has-text has-icon"
        disabled=""
        id="form-file-upload-field_key"
        type="button"
      >
        <svg
          class="gridicon gridicons-add-outline"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm5 9h-4V7h-2v4H7v2h4v4h2v-4h4v-2z"
            />
          </g>
        </svg>
        Upload file
      </button>
      <input
        accept=".pdf, image/png, image/jpeg"
        data-testid="form-file-upload-input"
        style="display: none;"
        type="file"
      />
    </div>
  </div>
</div>
`;

exports[`FileUploadControl renders upload done state 1`] = `
<div>
  <div
    class="components-drop-zone"
    data-is-drop-zone="true"
  >
    <div
      class="components-drop-zone__content"
    >
      <div
        class="components-drop-zone__content-inner"
      >
        <svg
          aria-hidden="true"
          class="components-drop-zone__content-icon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.5 15v3.5H13V6.7l4.5 4.1 1-1.1-6.2-5.8-5.8 5.8 1 1.1 4-4v11.7h-6V15H4v5h16v-5z"
          />
        </svg>
        <span
          class="components-drop-zone__content-text"
        >
          Drop files to upload
        </span>
      </div>
    </div>
  </div>
  <div
    class="file-upload"
  >
    <div
      class="components-form-file-upload"
    >
      <button
        class="components-button is-success is-secondary has-text has-icon"
        id="form-file-upload-field_key"
        type="button"
      >
        <svg
          class="gridicon gridicons-checkmark"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M9 19.414l-6.707-6.707 1.414-1.414L9 16.586 20.293 5.293l1.414 1.414z"
            />
          </g>
        </svg>
        Upload file
      </button>
      <input
        accept=".pdf, image/png, image/jpeg"
        data-testid="form-file-upload-input"
        style="display: none;"
        type="file"
      />
    </div>
    <span
      class="upload-message"
    >
      file.pdf
    </span>
    <button
      aria-label="Remove file"
      class="components-button delete-uploaded-file-button has-icon"
      type="button"
    >
      <svg
        class="gridicon gridicons-trash"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M6.187 8h11.625l-.695 11.125A2 2 0 0115.121 21H8.879a2 2 0 01-1.996-1.875L6.187 8zM19 5v2H5V5h3V4a2 2 0 012-2h4a2 2 0 012 2v1h3zm-9 0h4V4h-4v1z"
          />
        </g>
      </svg>
    </button>
  </div>
</div>
`;

exports[`FileUploadControl renders upload failed state 1`] = `
<div>
  <div
    class="components-drop-zone"
    data-is-drop-zone="true"
  >
    <div
      class="components-drop-zone__content"
    >
      <div
        class="components-drop-zone__content-inner"
      >
        <svg
          aria-hidden="true"
          class="components-drop-zone__content-icon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.5 15v3.5H13V6.7l4.5 4.1 1-1.1-6.2-5.8-5.8 5.8 1 1.1 4-4v11.7h-6V15H4v5h16v-5z"
          />
        </svg>
        <span
          class="components-drop-zone__content-text"
        >
          Drop files to upload
        </span>
      </div>
    </div>
  </div>
  <div
    class="file-upload"
  >
    <div
      class="components-form-file-upload"
    >
      <button
        class="components-button is-secondary is-destructive has-text has-icon"
        id="form-file-upload-field_key"
        type="button"
      >
        <svg
          class="gridicon gridicons-add-outline"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm5 9h-4V7h-2v4H7v2h4v4h2v-4h4v-2z"
            />
          </g>
        </svg>
        Upload file
      </button>
      <input
        accept=".pdf, image/png, image/jpeg"
        data-testid="form-file-upload-input"
        style="display: none;"
        type="file"
      />
    </div>
    <span
      class="upload-message is-destructive"
    >
      Error message
    </span>
  </div>
</div>
`;
